src/connector/src/sink/sqlserver.rs

```diff
@@ -495,28 +495,43 @@ impl SinkWriter for SqlServerSinkWriter {
     }
 }
 
-struct SqlClient {
-    client: Client<tokio_util::compat::Compat<TcpStream>>,
+#[derive(Debug)]
+pub struct SqlServerClient {
+    pub inner_client: Client<tokio_util::compat::Compat<TcpStream>>,
 }
 
-impl SqlClient {
+impl SqlServerClient {
     async fn new(msconfig: &SqlServerConfig) -> Result<Self> {
         let mut config = Config::new();
         config.host(&msconfig.host);
         config.port(msconfig.port);
         config.authentication(AuthMethod::sql_server(&msconfig.user, &msconfig.password));
         config.database(&msconfig.database);
         config.trust_cert();
+        Self::new_with_config(config).await
+    }
+
+    pub async fn new_with_config(mut config: Config) -> Result<Self> {
+        let tcp = TcpStream::connect(config.get_addr()).await.unwrap();
+        tcp.set_nodelay(true).unwrap();
```

Comment:
Should preserve the error context as before and throw the error instead of `unwrap` directly.

Reply:
updated